evalml/automl/automl_search.py

```diff
@@ -748,18 +702,64 @@ def _add_result(self, trained_pipeline, parameters, training_time, cv_data, cv_s
+    def _evaluate_pipelines(self, current_pipeline_batch, X, y, baseline=False, search_iteration_plot=None):
+        current_batch_pipeline_scores = []
+        add_single_pipeline = False
+        if isinstance(current_pipeline_batch, PipelineBase):
+            current_pipeline_batch = [current_pipeline_batch]
+            add_single_pipeline = True
+
+        while len(current_pipeline_batch) > 0 and (add_single_pipeline or baseline or self._check_stopping_condition(self._start)):
+            pipeline = current_pipeline_batch.pop()
+            try:
+                parameters = pipeline.parameters
+                logger.debug('Evaluating pipeline {}'.format(pipeline.name))
+                logger.debug('Pipeline parameters: {}'.format(parameters))
+
+                if self.start_iteration_callback:
+                    self.start_iteration_callback(pipeline.__class__, parameters, self)
+                desc = f"{pipeline.name}"
+                if len(desc) > self._MAX_NAME_LEN:
+                    desc = desc[:self._MAX_NAME_LEN - 3] + "..."
+                desc = desc.ljust(self._MAX_NAME_LEN)
+
+                if not add_single_pipeline:
+                    update_pipeline(logger, desc, len(self._results['pipeline_results']) + 1, self.max_iterations,
+                                    self._start, 1 if baseline else self._automl_algorithm.batch_number, self.show_batch_output)
+
+                evaluation_results = self._compute_cv_scores(pipeline, X, y)
+                parameters = pipeline.parameters
+
+                if baseline:
+                    self._baseline_cv_scores = self._get_mean_cv_scores_for_all_objectives(evaluation_results["cv_data"])
+
+                logger.debug('Adding results for pipeline {}\nparameters {}\nevaluation_results {}'.format(pipeline.name, parameters, evaluation_results))
+                self._add_result(trained_pipeline=pipeline,
+                                 parameters=parameters,
+                                 training_time=evaluation_results['training_time'],
+                                 cv_data=evaluation_results['cv_data'],
+                                 cv_scores=evaluation_results['cv_scores'])
+                logger.debug('Adding results complete')
+
+                score = evaluation_results['cv_score_mean']
+                score_to_minimize = -score if self.objective.greater_is_better else score
+                current_batch_pipeline_scores.append(score_to_minimize)
+
+                if not baseline and not add_single_pipeline:
+                    self._automl_algorithm.add_result(score_to_minimize, pipeline)
+
+                if search_iteration_plot:
+                    search_iteration_plot.update()
+
+                if add_single_pipeline:
+                    add_single_pipeline = False
+
+            except KeyboardInterrupt:
```

Comment:
From what I can tell there isn't any change to the keyboard interrupt feature! What do you think @christopherbunn ?

Reply:
I think technically if the user terminates the search while the next batch is being generated, it wouldn't get caught by this KeyboardInterrupt. In practice, getting the next batch takes so little time it's very unlikely that this will occur.